Optimizer/LexicaseOptimizer/LexicaseOptimizer.cpp

```diff
@@ -0,0 +1,268 @@
+		maxScores.push_back(optimizeFormulasMTs[fIndex]->eval(population[0]->dataMap, PT)[0]);
+		minScores.push_back(maxScores.back());
+	}
+
+	killList.clear(); // will hold orgs which will be cleaned up after archive
+	scoresHaveDelta = false; // if all scores are the same, then random selection will result
+
+	// get all scores and record to dataMap
+	scores.clear();
+	for (size_t i = 0; i < population.size(); i++) {
+		// for each org in population
+		killList.insert(population[i]); // add to kill list so that they are deleted in cleanup step
+		scores.push_back({});
+		for (size_t fIndex = 0; fIndex < optimizeFormulasMTs.size(); fIndex++) {
+			scores[i].push_back(optimizeFormulasMTs[fIndex]->eval(population[i]->dataMap, PT)[0]);
+			aveScores[fIndex] += scores.back().back();
+			if (recordOptimizeValues){
+				population[i]->dataMap.set(scoreNames[fIndex], scores.back().back());
+			}
+			if (scores[i][fIndex] > maxScores[fIndex]) {
+				maxScores[fIndex] = scores[i][fIndex];
+				scoresHaveDelta = true;
+			}
+			if (scores[i][fIndex] < minScores[fIndex]) {
+				minScores[fIndex] = scores[i][fIndex];
+				scoresHaveDelta = true;
+			}
+		}
+	}
+
+	for (size_t fIndex = 0; fIndex < optimizeFormulasMTs.size(); fIndex++) {
+		aveScores[fIndex] /= oldPopulationSize;
+	}
+	
+	// hold newly generated orgs - do not add to population until all have been selected
+	std::vector<std::shared_ptr<Organism>> newPopulation;
+
+	// temp vector used to pass tournamentSize randomly selected orgs to lexicase algorithm
+	std::vector<int> tournamentIndexList;
+
+	// initialize tournamentIndexList to whole population
+	for (size_t i = 0; i < population.size(); i++) {
+		tournamentIndexList.push_back(i);
+	}
+
+	std::shared_ptr<Organism> parent; // for asexual
+	std::vector<std::shared_ptr<Organism>> parents; // for sexual
+
+	while (nextPopulationSize < nextPopulationTargetSize) {
+		if (numberParents == 1) { // asexual reproduction
+			if (tournamentSize == -1) {
+				parent = population[lexiSelect(tournamentIndexList)];
+			}
+			else {
+				tournamentIndexList.clear();
+				for (int i = 0; i < tournamentSize; i++) {
+					tournamentIndexList.push_back(Random::getIndex(population.size()));
+				}
+				parent = population[lexiSelect(tournamentIndexList)];
+			}
+			newPopulation.push_back(parent->makeMutatedOffspringFrom(parent)); // push to population
+		}
+		else { // sexual reproduction
+			parents.clear();
+			while (static_cast<int>(parents.size()) < numberParents) {
+				if (tournamentSize == -1) {
+					parents.push_back(population[lexiSelect(tournamentIndexList)]);
+				}
+				else {
+					tournamentIndexList.clear();
+					for (int i = 0; i < tournamentSize; i++) {
+						tournamentIndexList.push_back(Random::getIndex(population.size()));
+					}
+					parents.push_back(population[lexiSelect(tournamentIndexList)]);
+				}
+			}
+			newPopulation.push_back(parents[0]->makeMutatedOffspringFromMany(parents));
+		}
+		nextPopulationSize++;
```

Comment:
I see the point here. But I don't want to change it so that if in the future I want to add the option to produce more then one offspring per parent(s) selection I can add that more easily.